tests/unit/services/flash-messages-test.js

```diff
@@ -205,6 +216,43 @@ module('FlashMessagesService', function(hooks) {
     assert.equal(get(service, 'queue').length, 2, 'it does not add duplicate messages to the queue');
   });
 
+  test('it does not add duplicate messages to the queue if flashInstance preventDuplicates is `true`', function(assert) {
+    set(service, 'defaultPreventDuplicates', false);
+    const messages = emberArray([ 'foo', 'foo', 'bar' ]);
+    const expectedResult = messages.uniq();
+    messages.forEach((message) => service.success(message, {preventDuplicates: true}));
+    const result = get(service, 'queue').mapBy('message');
+
+    assert.deepEqual(result, expectedResult, 'it does not add duplicate messages to the queue');
+    assert.equal(get(service, 'queue').length, 2, 'it does not add duplicate messages to the queue');
+  });
+
+  test('flashInstance preventDuplicates prefers instance preventDuplicates over global', function(assert) {
+    set(service, 'defaultPreventDuplicates', true);
+    const messages = emberArray([ 'foo', 'foo', 'bar' ]);
+    const expectedResult = [ 'foo', 'foo', 'bar' ];
+    messages.forEach((message) => service.success(message, {preventDuplicates: false}));
+    const result = get(service, 'queue').mapBy('message');
+
+    assert.deepEqual(result, expectedResult, 'it adds duplicate messages to the queue');
+    assert.equal(get(service, 'queue').length, 3, 'it adds duplicate messages to the queue');
+  });
+
+  test('it does adds duplicate messages where preventDuplicates is `true`', function(assert) {
```

Comment:
The test description here is a bit confusing to me. Aren't we saying it will honor the global default if an instance `preventDuplicates` is not set? i.e. we expect multiple `foo`'s but only one `bar` in this test case?

The test makes sense, I just find this wording a bit confusing

Reply:
Yes you're right.

I've adjusted the test title